src/main/java/com/microsoft/sqlserver/jdbc/SQLServerStatement.java

```diff
@@ -1038,6 +1038,16 @@ static String replaceMarkerWithNull(String sql) {
         }
     }
 
+    static boolean containsUnEscapedQuotes(String sql) {
+        // Look for bracketed strings to be ignored
+        Pattern p = Pattern.compile("\\[(.*?)\\]");
```

Comment:
I don't think this covers the other quoting character allowed for identifiers, i.e. the double quote (") character.

Also, don't think it covers identifiers like [a]]'].

Reply:
Hi @and-costea 

I will close the PR for now as it would require a strong parser implementation to validate all cases. We have similar parsing logic in the driver but it isn't generalized to be re-used in this case. I will keep the issue open to bring up that parser in effect in this case.